gradle-palantir-java-format/src/main/java/com/palantir/javaformat/gradle/NativeImageFormatProviderPlugin.java

```diff
@@ -0,0 +1,89 @@
+ */
+
+package com.palantir.javaformat.gradle;
+
+import com.google.common.base.Preconditions;
+import com.palantir.platform.Architecture;
+import com.palantir.platform.OperatingSystem;
+import java.util.Optional;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.artifacts.type.ArtifactTypeDefinition;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+
+public final class NativeImageFormatProviderPlugin implements Plugin<Project> {
+
+    private static final Logger log = Logging.getLogger(NativeImageFormatProviderPlugin.class);
+    static final String NATIVE_CONFIGURATION_NAME = "palantirJavaFormatNative";
+
+    @Override
+    public void apply(Project rootProject) {
+        Preconditions.checkState(
+                rootProject == rootProject.getRootProject(),
+                "May only apply com.palantir.java-format-provider to the root project");
+
+        if (!isNativeImageSupported()) {
+            log.info("Skipping native image configuration as it is not supported on this platform");
+            return;
+        }
+        String implementationVersion = JavaFormatExtension.class.getPackage().getImplementationVersion();
+        OperatingSystem operatingSystem = OperatingSystem.get();
+        rootProject.getConfigurations().register(NATIVE_CONFIGURATION_NAME, conf -> {
+            conf.setDescription("Internal configuration for resolving the palantir-java-format native image");
+            conf.setVisible(false);
+            conf.setCanBeConsumed(false);
+            conf.setCanBeResolved(true);
+            conf.defaultDependencies(deps -> {
+                deps.add(rootProject
+                        .getDependencies()
+                        .create(String.format(
+                                "com.palantir.javaformat:palantir-java-format-native:%s:nativeImage-%s_%s@%s",
+                                implementationVersion,
+                                operatingSystem.uiName(),
+                                Architecture.get().uiName(),
+                                getExtension(operatingSystem))));
+            });
+            conf.getAttributes().attribute(ArtifactTypeDefinition.ARTIFACT_TYPE_ATTRIBUTE, "executable-nativeImage");
+        });
+        rootProject.getDependencies().registerTransform(ExecutableTransform.class, transformSpec -> {
+            transformSpec
+                    .getFrom()
+                    .attribute(ArtifactTypeDefinition.ARTIFACT_TYPE_ATTRIBUTE, getExtension(operatingSystem));
+            transformSpec.getTo().attribute(ArtifactTypeDefinition.ARTIFACT_TYPE_ATTRIBUTE, "executable-nativeImage");
+        });
+    }
+
+    public static boolean isNativeImageSupported() {
+        OperatingSystem os = OperatingSystem.get();
+        return os.equals(OperatingSystem.LINUX_GLIBC)
+                || (os.equals(OperatingSystem.MACOS) && Architecture.get().equals(Architecture.AARCH64));
```

Comment:
I think it needs to be `aarch64 && (linux_glibc || macos)`, we don't support linux_glibc aarch64 either.

Reply:
we support both aarch64 & amd64 for linux: https://github.com/palantir/palantir-java-format/blob/develop/.circleci/config.yml#L22; and https://repo.maven.apache.org/maven2/com/palantir/javaformat/palantir-java-format-native/2.52.0/